library/std/src/sys_common/ucs2.rs

```diff
@@ -0,0 +1,131 @@
+//! Implementating of UCS-2 encoding. This is a lesser known subset of UTF-16 without surrogate
+//! pairs. It is used in UEFI
+#![allow(dead_code)]
+
+use crate::fmt;
+use crate::iter::FusedIterator;
+use crate::num::NonZeroU16;
+use crate::str;
+
+/// A struct to represent UCS-2 character
+#[unstable(feature = "ucs2", issue = "none")]
+#[repr(transparent)]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+pub struct Ucs2Char {
+    value: NonZeroU16,
+}
```

Comment:
I think it would be great to use the existing UTF-16 implementation if it does not cause any bugs. I can give it a try and run all the tests to see if it breaks anything.

Reply:
@thomcc What do you think about this?

Reply:
I'd prefer to use UTF-16, but as I've mentioned, I'm not knowledgable with UEFI, so perhaps @joshtriplett might have stronger feelings.

(Also sorry for the delay here, I should be able to another review pass next week)

Reply:
Well, I have switched to using UTF-16 for UEFI. This means I am now using `std::os::windows::ffi` and `std::sys::windows::os_str` instead of UEFI specific one. There have been no regressions, so I guess I will resolve this conversation for now.